src/StatefulERC712.sol

```diff
@@ -4,14 +4,14 @@ pragma solidity 0.8.23;
 
 import { IStatefulERC712 } from "./interfaces/IStatefulERC712.sol";
 
-import { ERC712 } from "./ERC712.sol";
+import { ERC712Domain } from "./ERC712Domain.sol";
 
 /// @title Stateful Extension for EIP-712 typed structured data hashing and signing with nonces.
 /// @dev   An abstract implementation to satisfy stateful EIP-712 with nonces.
-abstract contract StatefulERC712 is IStatefulERC712, ERC712 {
+abstract contract StatefulERC712 is IStatefulERC712, ERC712Domain {
     mapping(address account => uint256 nonce) internal _nonces; // Nonces for all signatures.
```

Comment:
This can be public if `function nonces` is not virtual. This is my fault for not having done this earlier.

Reply:
Fixed in this commit: https://github.com/MZero-Labs/common/pull/7/commits/c98a486dcf668a367f03459dc7be7114682a9244